mmocr/models/textdet/postprocess/wrapper.py

```diff
@@ -391,3 +393,169 @@ def textsnake_decode(preds,
             boundaries.append(boundary + [score])
 
     return boundaries
+
+
+def fcenet_decode(
+    preds,
+    fourier_degree,
+    reconstr_points,
+    scale,
+    tcl_tr_ratio,
+    text_repr_type='poly',
+    tr_thresh=0.8,
+    nms_thresh=0.1,
+):
+    """Decoding predictions of FCEnet to instances.
+
+    Args:
+        preds (list(Tensor)): The head output tensors.
+        fourier_degree (int): The maximum fourier transform degree k
+        reconstr_points (int): The points number of the polygon reconstructed from predicted
+            fourier coefficients.
+        scale (int): The downsample scale of preds
+        tcl_tr_ratio (float): The ratio to calculate final score.
+        text_repr_type (str): boundary encoding type 'poly' or 'quad'.
+        tr_thresh (float) : The threshold used to filter out the final
+            candidates.
+        nms_thresh (float) : The threshold of nms.
+
+    Returns:
+        boundaries (list[list[float]]): The instance boundary and its
+        instance confidence list.
+    """
+    assert isinstance(preds, list)
+    assert len(preds) == 2
+    assert text_repr_type == 'poly'
+
+    cls_pred = preds[0][0]
+    tr_pred = cls_pred[0:2].softmax(dim=0).data.cpu().numpy()
+    tcl_pred = cls_pred[2:].softmax(dim=0).data.cpu().numpy()
+
+    reg_pred = preds[1][0].permute(1, 2, 0).data.cpu().numpy()
+    x_pred = reg_pred[:, :, :2 * fourier_degree + 1]
+    y_pred = reg_pred[:, :, 2 * fourier_degree + 1:]
+
+    score_pred = tr_pred[1] * (tcl_pred[1]**tcl_tr_ratio)
+    tr_pred_mask = (score_pred) > tr_thresh
+    tr_mask = fill_hole(tr_pred_mask)
+
+    tr_contours, _ = cv2.findContours(
+        tr_mask.astype(np.uint8), cv2.RETR_TREE,
+        cv2.CHAIN_APPROX_SIMPLE)  # opencv4
+
+    mask = np.zeros_like(tr_mask)
+    e_matrix = generate_e_matrix(reconstr_points, fourier_degree)
+    boundaries = []
+    for cont in tr_contours:
+        deal_map = mask.copy().astype(np.int8)
+        cv2.drawContours(deal_map, [cont], -1, 1, -1)
+
+        text_map = score_pred * deal_map
+        polygons = contour_transfor_inv(fourier_degree, x_pred, y_pred, text_map, e_matrix,
+                                        scale)
+        polygons = poly_nms(polygons, nms_thresh)
+        boundaries = boundaries + polygons
+
+    boundaries = poly_nms(boundaries, nms_thresh)
+    return boundaries
+
+
+def poly_nms(polygons, threshold):
```

Comment:
what's the different between poly_nms between nms?
is it necessary?

Reply:
function nms in mmdet only deal with rectangle bbox, which is not suited for polygon.